src/test/ui/const-generics/issues/issue-90318.stderr

```diff
@@ -1,29 +1,51 @@
-error[E0277]: can't compare `TypeId` with `_` in const contexts
-  --> $DIR/issue-90318.rs:14:28
+error: overly complex generic constant
+  --> $DIR/issue-90318.rs:15:8
    |
 LL |     If<{ TypeId::of::<T>() != TypeId::of::<()>() }>: True,
-   |                            ^^ no implementation for `TypeId == _`
+   |        ^^-----------------^^^^^^^^^^^^^^^^^^^^^^^^
+   |          |
+   |          borrowing is not supported in generic constants
    |
-   = help: the trait `~const PartialEq<_>` is not implemented for `TypeId`
-note: the trait `PartialEq<_>` is implemented for `TypeId`, but that implementation is not `const`
-  --> $DIR/issue-90318.rs:14:28
+   = help: consider moving this anonymous constant into a `const` function
+   = note: this operation may be supported in the future
+
+error[E0015]: cannot call non-const operator in constants
+  --> $DIR/issue-90318.rs:15:10
    |
 LL |     If<{ TypeId::of::<T>() != TypeId::of::<()>() }>: True,
-   |                            ^^
+   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+note: impl defined here, but it is not `const`
+  --> $SRC_DIR/core/src/any.rs:LL:COL
+   |
+LL | impl const PartialEq for TypeId {
```

Comment:
Isn't this a bit weird? “*impl defined here but it is not const*” but it is const as shown?

Reply:
Yes that is because this test doesn't use the `const_trait_impl` feature flag.
That always leads to this error if used this way.

The same error happens in `src/test/ui/consts/const_cmp_type_id.rs` if `#![feature(const_trait_impl)]` is removed, this is probably more of an issue in `const_trait_impl` than here.